dali/c_api/c_api_test.cc

```diff
@@ -704,26 +740,32 @@ void TestForceFlagRun(bool ext_src_no_copy, unsigned int flag_to_test) {
                      device_id, false, prefetch_queue_depth, prefetch_queue_depth,
                      prefetch_queue_depth, false);
 
-  std::vector<TensorList<TypeParam>> data;
-  data.resize(prefetch_queue_depth);
+  std::vector<std::shared_ptr<uint8_t>> data;
+  data.reserve(prefetch_queue_depth);
+  for (int i = 0; i < prefetch_queue_depth; i++) {
+    data.push_back(AllocBuffer<TypeParam>(num_elems * sizeof(uint8_t), false));
+  }
+  std::vector<TensorList<TypeParam>> input_wrapper(3);
```

Comment:
Maybe it is better to use TensorVector unless TensorList is the data structure we are reworking towards?

Reply:
Hmm, it should be 
```suggestion
  std::vector<TensorList<TypeParam>> input_wrapper(prefetch_queue_depth);
```
but other then that I just kept the TensorList. What we will use is up to be decided, but I've heard voices to keep the TensorList to match what we have in Python. But we will see.